src/rgw/rgw_bucket.cc

```diff
@@ -423,7 +423,22 @@ int rgw_bucket_set_attrs(RGWRados *store, RGWBucketInfo& bucket_info,
 
   ::encode(bucket_info, bl);
 
-  return rgw_bucket_instance_store_info(store, key, bl, false, &attrs, objv_tracker, real_time());
+  auto r = rgw_bucket_instance_store_info(store, key, bl, false,
+					  &attrs, objv_tracker, real_time());
+  if (r == -ECANCELED) {
+    for (auto i = 0; i < 15; ++i || r != -ECANCELED) {
```

Comment:
@adamemerson could it be that the condition here is broken?

Reply:
correct it can be broken